### ISSUE TITLE
Module: Line numbers for unexpected error instead of character position

### ISSUE THREAD

**Issue by Mike-Dax** [microsoft/TypeScript-Website#2051]
**Improvement Idea:**

When unexpected errors occur, `@typescript/twoslash` presents a string to the developer:

https://github.com/microsoft/TypeScript-Website/blob/1ffeded798ca705d8893c39f4cfd780889784900/packages/ts-twoslasher/src/validation.ts#L42

With non-trivial code examples and multiple errors, it's hard for the human to pinpoint which line in the code was actually the problem from this character position information.

I incorrectly assumed `e.start` was a line number, another co-worker assumed it was the 'line position from the start of the  markdown file' (we're using shiki-twoslash with a static site generator to generate code snippets with better tooltips).

I would suggest `${e.start}` be changed to the line number instead.

I'm happy to open a PR if this is acceptable.

---

From a cursory look, this doesn't seem to be trivially available from the error. Would something like this be sane to calculate the line number?

```ts
const codeUpToError = originalCode.slice(0, e.start ?? 0)
const lineNumber = (codeUpToError.match(/\n/g)||[]).length
```

**Comment by orta** [comment]
Yeah, `line number:character` is better I agree! (it matches how we do it in tsc)
